### PR TITLE
Add 8kHz training test coverage

### LINKED ISSUE #2638: Add 8kHz training test

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #2638

### CODE REVIEW

**Comment by reuben** [comment]
Did you forget to `git add` a file? The definition of `﻿set_ldc_sample_filename` is not here.

**Comment by reuben** [comment]
Nevermind, GitHub was just hiding it.

File: taskcluster/tc-train-tests.sh [hunk 1]
```diff
@@ -56,11 +56,28 @@ decoder_pkg_url=${DECODER_ARTIFACTS_ROOT}/${decoder_pkg}
 
 LD_LIBRARY_PATH=${PY37_LDPATH}:$LD_LIBRARY_PATH pip install --verbose --only-binary :all: ${PY37_SOURCE_PACKAGE} ${decoder_pkg_url} | cat
 
+# Prepare correct arguments for training
+case "${bitrate}" in
+    8k)
+        sample_rate=8000
+	sample_name='LDC93S1_pcms16le_1_8000.wav'
```
Comment (reuben) [inline taskcluster/tc-train-tests.sh]: nit: mixed indentation

File: taskcluster/test-cpp_16k-linux-amd64-opt.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+build:
+  template_file: test-linux-opt-base.tyml
+  dependencies:
+    - "linux-amd64-cpu-opt"
+    - "test-training_16k-linux-amd64-py36m-opt"
+  test_model_task: "test-training_16k-linux-amd64-py36m-opt"
+  test_model_task: "test-training_16k-linux-amd64-py36m-opt"
```
Comment (reuben) [inline taskcluster/test-cpp_16k-linux-amd64-opt.yml]: Duplicated key.

File: taskcluster/test-electronjs_v7.0-darwin-amd64-opt.yml [hunk 3]
```diff
@@ -2,12 +2,13 @@ build:
   template_file: test-darwin-opt-base.tyml
   dependencies:
     - "darwin-amd64-cpu-opt"
-    - "test-training_upstream-linux-amd64-py35m-opt"
```
Comment (reuben) [inline taskcluster/test-electronjs_v7.0-darwin-amd64-opt.yml]: Thanks for fixing.

**Comment by lissyx** [comment]
> Looks good, I left a couple of minor comments inline. Should we also have 8kHz tests for the Python/NodeJS clients?

We could but I'm unsure about the actual value.

**Comment by reuben** [comment]
All of the resampling logic is replicated in each client. For example, here's a few differences in the logic between the C++, Python and Node.JS clients:

- C++ and Node.JS clients always feeds the data through SoX, only warns about erratic speech recognition when input sample rate is **LESS THAN** model sample rate.
- Python client only feeds the data through SoX when the sample rate differs from the model, and warns about erratic speech recognition when input rate **NOT EQUAL TO** model sample rate.

**Comment by lissyx** [comment]
> All of the resampling logic is replicated in each client. For example, here's a few differences in the logic between the C++, Python and Node.JS clients:
> 
>     * C++ and Node.JS clients always feeds the data through SoX, only warns about erratic speech recognition when input sample rate is **LESS THAN** model sample rate.
> 
>     * Python client only feeds the data through SoX when the sample rate differs from the model, and warns about erratic speech recognition when input rate **NOT EQUAL TO** model sample rate.

Yeah, let's do as suggested on IRC: linux amd64 only, TF/TFLite for all of C++, Python and NodeJS/ElectronJS.

**Pull request merged**